src/main/java/com/google/sps/servlets/TasksServlet.java

```diff
@@ -52,6 +61,71 @@ public TasksServlet(
     this.tasksClientFactory = tasksClientFactory;
   }
 
+  private List<Task> getTasks(TasksClient tasksClient) throws IOException {
+    List<TaskList> taskLists = tasksClient.listTaskLists();
+    List<Task> tasks = new ArrayList<>();
+    for (TaskList taskList : taskLists) {
+      tasks.addAll(tasksClient.listTasks(taskList));
+    }
+    return tasks;
+  }
+
+  private List<String> getTaskListTitles(List<TaskList> taskLists) throws IOException {
+    return taskLists.stream().map(taskList -> taskList.getTitle()).collect(Collectors.toList());
+  }
+
+  private int getTasksToComplete(List<Task> tasks) {
+    // getHidden is defined for incomplete tasks
+    List<Task> tasksToComplete =
+        tasks.stream().filter(task -> task.getHidden() == null).collect(Collectors.toList());
+    return tasksToComplete.size();
+  }
+
+  private int getTasksDueToday(List<Task> tasks) {
+    String today = LocalDate.now().toString();
+    List<Task> tasksDueToday =
+        tasks.stream()
+            .filter(task -> task.getDue() != null && task.getDue().contains(today))
+            .collect(Collectors.toList());
+    return tasksDueToday.size();
+  }
+
+  private int getTasksCompletedToday(List<Task> tasks) {
+    ZoneId zoneId = ZoneId.systemDefault();
+    Instant startOfDay = LocalDate.now(zoneId).atStartOfDay(zoneId).toInstant();
+    Instant endOfDay = LocalDate.now(zoneId).plusDays(1).atStartOfDay(zoneId).toInstant();
+    List<Task> tasksCompletedToday = new ArrayList<>();
+    for (Task task : tasks) {
+      // getHidden is defined for completed tasks
+      if (task.getHidden() != null) {
+        Instant completionDateTime = ZonedDateTime.parse(task.getUpdated()).toInstant();
+        if (completionDateTime.isAfter(startOfDay) && completionDateTime.isBefore(endOfDay)) {
+          tasksCompletedToday.add(task);
+        }
+      }
+    }
+    return tasksCompletedToday.size();
+  }
+
+  private int getTasksOverdue(List<Task> tasks) {
+    ZoneId zoneId = ZoneId.systemDefault();
+    String zoneOffset = zoneId.getRules().getOffset(Instant.now()).toString();
+    Instant endOfDay = LocalDate.now(zoneId).plusDays(1).atStartOfDay(zoneId).toInstant();
+    List<Task> tasksOverdue = new ArrayList<>();
```

Comment:
If you only care about the size, maybe you should just have a counter?

Reply:
Agreed! I have done the same for `getTasksCompletedToday`. Should I do the same for `getTasksToComplete` and `getTasksDueToday`? This would entail using an explicit for loop instead of streams. I imagine that it would be better to use an explicit for loop + save some memory rather than using streams but I could be wrong.

Reply:
Maybe you could still use [.count()](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#count--)?

Reply:
Didn't know about `.count()` - it's perfect, thanks!

Reply:
`RESOLVED` #105